res/controllers/Arturia-KeyLab-Mk1-scripts.js

```diff
@@ -0,0 +1,1007 @@
+"use strict";
+
+// eslint-disable-next-line no-var
+var KeyLabMk1;
+(function(KeyLabMk1) {
+    // https://manual.mixxx.org/2.5/en/chapters/appendix/mixxx_controls#control-[ChannelN]-key
+    const keyNums = [
+        "C",
+        "Db",
+        "D",
+        "Eb",
+        "E",
+        "F",
+        "F#/Gb",
+        "G",
+        "Ab",
+        "A",
+        "Bb",
+        "B",
+        "Cm",
+        "C#m",
+        "Dm",
+        "D#m/Ebm",
+        "Em",
+        "Fm",
+        "F#m",
+        "Gm",
+        "G#m",
+        "Am",
+        "Bbm",
+        "Bm",
+    ];
+
+    // https://manual.mixxx.org/2.5/en/chapters/appendix/mixxx_controls#control-[Library]-sort_column
+    const sortCols = [
+        "Artist",
+        "Title",
+        "Album",
+        "Album Artist",
+        "Year",
+        "Genre",
+        "Composer",
+        "Grouping",
+        "Tracknumber",
+        "Filetype",
+        "Native Location",
+        "Comment",
+        "Duration",
+        "Bitrate",
+        "BPM",
+        "Replay Gain",
+        "Datetime Added",
+        "Times Played",
+        "Rating",
+        "Key",
+        "Preview",
+        "Cover Art",
+        "Position",
+        "Play ID",
+        "Location",
+        "Filename",
+        "Modified Time",
+        "Creation Time",
+        "Sample Rate",
+        "Track Color",
+        "Last Played",
+    ];
+
+    // Menu items that can be controlled with the parameter and value encoders.
+    const menu = [{
+        name: "Key",
+        rotate: {
+            inKey: "key",
+            max: 24,
+            outputValue: (value) => {
+                return keyNums[script.posMod(value - 1, keyNums.length)];
+            },
```

Comment:
isn't this just `outValueScale`?

Reply:
Oops, yup, that's basically doing the same thing. Fixed.